### ISSUE TITLE
[Support]: Frigate not starting after moving to docker 0.13.1 from 0.12

### ISSUE THREAD

**Issue by briodan** [blakeblackshear/frigate#9600]
### Describe the problem you are having

container does not start, complaining that it cant connect to upstream client

docker compose file:

```
version: '2.4'
networks:
    iot-net:
        external: true
services:
    rtsptoweb:
        container_name: frigate
        restart: unless-stopped
        image: ghcr.io/blakeblackshear/frigate:stable
        shm_size: "250mb"
        networks:
            iot-net:
                ipv4_address: 10.10.10.251
        volumes:
            - type: bind
              source: /dockershare/frigate/clips
              target: /media/frigate/clips
            - type: bind
              source: /dockershare/frigate/recordings
              target: /media/frigate/recordings
            - type: bind
              source: /dockershare/frigate/config
              target: /config 
            - /etc/timezone:/etc/timezone:ro
            - /etc/localtime:/etc/localtime:ro
        devices:
            - /dev/apex_0:/dev/apex_0
```

### Version

0.13.1

### Frigate config file

```yaml
mqtt:
  host: 10.10.10.225
  user: homeassistant
  password: XXXXXXXXX
birdseye:
  enabled: True
  mode: continuous
detectors:
#  cpu:
#    type: cpu
  coral_pci:
    type: edgetpu
    device: pci
  
cameras:
  camera_front: # <------ Name the camera
    ffmpeg:
      inputs:
        - path: rtsp://10.10.10.252:8554/camera_front # <----- Update for your camera
          input_args: preset-rtsp-restream
          roles:
            - detect
            - rtmp
    rtmp:
      enabled: False # <-- RTMP should be disabled if your stream is not H264
    detect:
      width: 704 # <---- update for your camera's resolution
      height: 480 # <---- update for your camera's resolutio
    objects:
      filters:
        car:
          mask:
            - 522,480,704,480,704,442,520,439
      track:
        - person
        - dog
        - cat
        - car
    zones:
      driveway_left:
        coordinates: 188,59,318,58,261,356,0,322
      driveway_person:
        coordinates: 501,66,215,45,0,313,0,480,704,480,704,357
      driveway_right:
        coordinates: 639,358,512,67,353,58,303,370
    snapshots:
      enabled: true
      clean_copy: true
      timestamp: true
      bounding_box: true
      crop: false
      retain:
        default: 2
        objects:
          person: 10
          car: 10
    record:
      enabled: True
      events:
        retain:
          default: 2
          mode: all
    mqtt:
      crop: false
  camera_front_door: # <------ Name the camera
    ffmpeg:
      inputs:
         - path: rtsp://10.10.10.252:8554/camera_front_door # <----- Update for your camera
           input_args: preset-rtsp-restream
           roles:
             - detect
             - rtmp
    rtmp:
      enabled: False # <-- RTMP should be disabled if your stream is not H264
    detect:
      width: 640 # <---- update for your camera's resolution
      height: 480 # <---- update for your camera's resolutio
    objects:
      filters:
        person:
          mask:
            - 356,255,354,286,328,291,331,260
      track:
        - person
        - dog
        - cat
    snapshots:
      enabled: true
      clean_copy: true
      timestamp: true
      bounding_box: true
      crop: false
      retain:
        default: 2
        objects:
          person: 10
    record:
      enabled: True
      events:
        retain:
          default: 2
          mode: all
  camera_back_door: # <------ Name the camera
    ffmpeg:
      inputs:
         - path: rtsp://10.10.10.252:8554/camera_back_door # <----- Update for your camera
           input_args: preset-rtsp-restream
           roles:
             - detect
             - rtmp
    rtmp:
      enabled: False # <-- RTMP should be disabled if your stream is not H264
    detect:
      width: 640 # <---- update for your camera's resolution
      height: 480 # <---- update for your camera's resolutio
    objects:
      track:
        - person
        - dog
        - cat
    snapshots:
      enabled: true
      clean_copy: true
      timestamp: true
      bounding_box: true
      crop: false
      retain:
        default: 2
        objects:
          person: 10
    record:
      enabled: True
      events:
        retain:
          default: 2
          mode: all
  camera_skunk: # <------ Name the camera
    ffmpeg:
      inputs:
        - path: rtsp://10.10.10.252:8554/camera_skunk # <----- Update for your camera
          input_args: preset-rtsp-restream
          roles:
            - detect
            - rtmp
    rtmp:
      enabled: False # <-- RTMP should be disabled if your stream is not H264
    detect:
      width: 1920 # <---- update for your camera's resolution
      height: 1080 # <---- update for your camera's resolutio
    objects:
      track:
        - person
        - dog
        - cat
    snapshots:
      enabled: true
      clean_copy: true
      timestamp: true
      bounding_box: true
      crop: false
      retain:
        default: 2
        objects:
          person: 10
          car: 10
    record:
      enabled: True
      events:
        retain:
          default: 2
          mode: all
  camera_garage: # <------ Name the camera
    ffmpeg:
      inputs:
        - path: rtsp://10.10.10.252:8554/camera_garage # <----- Update for your camera
          input_args: preset-rtsp-restream
          roles:
            - detect
            - rtmp
    rtmp:
      enabled: False # <-- RTMP should be disabled if your stream is not H264
    detect:
      width: 960 # <---- update for your camera's resolution
      height: 540 # <---- update for your camera's resolutio
    objects:
      track:
        - person
        - dog
        - cat
    snapshots:
      enabled: true
      clean_copy: true
      timestamp: true
      bounding_box: true
      crop: false
      retain:
        default: 2
        objects:
          person: 10
          car: 10
    record:
      enabled: True
      events:
        retain:
          default: 2
          mode: all
```

### Relevant log output

```shell
s6-rc: info: service s6rc-fdholder: starting
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-fdholder successfully started
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service log-prepare: starting
s6-rc: info: service log-prepare successfully started
s6-rc: info: service nginx-log: starting
s6-rc: info: service go2rtc-log: starting
s6-rc: info: service frigate-log: starting
s6-rc: info: service nginx-log successfully started
s6-rc: info: service go2rtc-log successfully started
s6-rc: info: service go2rtc: starting
s6-rc: info: service frigate-log successfully started
2024-02-02 08:11:12.960641728  [INFO] Preparing new go2rtc config...
s6-rc: info: service go2rtc successfully started
s6-rc: info: service go2rtc-healthcheck: starting
s6-rc: info: service frigate: starting
2024-02-02 08:11:12.978132990  [INFO] Preparing Frigate...
2024-02-02 08:11:12.992507710  [INFO] Starting Frigate...
s6-rc: info: service frigate successfully started
s6-rc: info: service nginx: starting
2024-02-02 08:11:13.004025063  [INFO] Starting NGINX...
s6-rc: info: service go2rtc-healthcheck successfully started
s6-rc: info: service nginx successfully started
s6-rc: info: service legacy-services: starting
s6-rc: info: service legacy-services successfully started
2024-02-02 08:11:13.260487523  [INFO] Starting go2rtc...
2024-02-02 08:11:13.329249217  08:11:13.329 INF go2rtc version 1.8.4 linux/amd64
2024-02-02 08:11:13.329701048  08:11:13.329 INF [rtsp] listen addr=:8554
2024-02-02 08:11:13.330041049  08:11:13.329 INF [webrtc] listen addr=:8555
2024-02-02 08:11:13.330237215  08:11:13.330 INF [api] listen addr=:1984
2024-02-02 08:11:13.872335639  2024/02/02 08:11:13 [error] 134#134: *1 connect() failed (111: Connection refused) while connecting to upstream, client: 10.10.10.225, server: , request: "GET /api/stats HTTP/1.1", upstream: "http://127.0.0.1:5001stats", host: "10.10.10.251:5000"
2024-02-02 08:11:14.133509058  [2024-02-02 08:11:14] frigate.app                    INFO    : Starting Frigate (0.13.1-34fb1c2)
s6-rc: info: service legacy-services: stopping
s6-rc: info: service legacy-services successfully stopped
s6-rc: info: service nginx: stopping
s6-rc: info: service go2rtc-healthcheck: stopping
2024-02-02 08:11:14.168125128  [INFO] The go2rtc-healthcheck service exited with code 256 (by signal 15)
s6-rc: info: service go2rtc-healthcheck successfully stopped
2024-02-02 08:11:14.219167807  [2024-02-02 08:11:14] peewee_migrate.logs            INFO    : Starting migrations
2024-02-02 08:11:14.223056125  [INFO] Service NGINX exited with code 0 (by signal 0)
s6-rc: info: service nginx successfully stopped
s6-rc: info: service nginx-log: stopping
s6-rc: info: service frigate: stopping
2024-02-02 08:11:14.247673193  [INFO] Service Frigate exited with code 256 (by signal 15)
s6-rc: info: service nginx-log successfully stopped
s6-rc: info: service frigate successfully stopped
s6-rc: info: service go2rtc: stopping
s6-rc: info: service frigate-log: stopping
2024-02-02 08:11:14.269611434  exit with signal: terminated
2024-02-02 08:11:14.273372294  [INFO] The go2rtc service exited with code 0 (by signal 0)
s6-rc: info: service go2rtc successfully stopped
s6-rc: info: service go2rtc-log: stopping
s6-rc: info: service frigate-log successfully stopped
s6-rc: info: service go2rtc-log successfully stopped
s6-rc: info: service log-prepare: stopping
s6-rc: info: service s6rc-fdholder: stopping
s6-rc: info: service log-prepare successfully stopped
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service s6rc-fdholder successfully stopped
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped
```

### FFprobe output from your camera

```shell
N/A
```

### Frigate stats

```json
N/A
```

### Operating system

Debian

### Install method

Docker Compose

### Coral version

M.2

### Network connection

Wired

### Camera make and model

N/A

**Comment by NickM-27** [comment]
it's not stopping due to the connection that is just an nginx log, there may be some permission issue on the config dir like a few others have seen

**Comment by briodan** [comment]
The exact same config, permissions and all had been running fine on 0.12 for months.

only thing I did is pull a new docker image.

**Comment by briodan** [comment]
permissions on the directories
![image](https://github.com/blakeblackshear/frigate/assets/41130293/2b5483fe-20d2-4468-bbe6-b8a3df078623)

permissions on the files in the config directory
![image](https://github.com/blakeblackshear/frigate/assets/41130293/9de4de7c-b824-4157-bda5-2702a171437d)

**Comment by GFBsoul** [comment]
I also had this problem. I ended up deploying a new container with new storage directories. Started it with the dummy config from https://docs.frigate.video/guides/getting_started

I then copied my old config in the web interface and it seems to be working now.

**Comment by traxx34** [comment]
I had this problem as well. I eliminated 

```
birdseye:
  enabled: true
  restream: true
```

and it now starts. I have lost the birdseye view as there is no camera entity for it.

**Comment by NickM-27** [comment]
> the ability to store the database on a network share but that looks to not be the case at least for me.

The change was done so that users would not store the DB on a network share. That's the main point.

**Comment by briodan** [comment]
> > the ability to store the database on a network share but that looks to not be the case at least for me.
> 
> The change was done so that users would not store the DB on a network share. That's the main point.

Thanks 

That's not at all how I read/understood the release notes. 
I guess it makes sense that you can now just map the media folder instead of each subfolder.

**Comment by Zetto12** [comment]
> I had this problem as well. I eliminated
> 
> ```
> birdseye:
>   enabled: true
>   restream: true
> ```
> 
> and it now starts. I have lost the birdseye view as there is no camera entity for it.

I commented out the Birdseye settings but it made no difference to me. It's still not starting.

I have the same problem. All I did was re-pull version 0.13.1 using Portainer (an update from Version 0.12 which ran without issues for months).

My config folder is stored locally. I only have the media folder on a network share mounted using fstab. The logfile doesn't give a clear indication what the issue is (as far as I can see) so I will probably just roll back.

**Comment by NickM-27** [comment]
@Zetto12 at the very least you should include your logs so others can see

**Comment by gnlgrim** [comment]
@briodan 
 /dockershare/frigate/config/config.yml:/config/config.yml

This saved my life.
I do not know why this is not documented in the notes. 
I had it as per below 
- /path/to/your/config:/config   ( as per documentation )
WHICH DOES NOT WORK.

Thanks you helped me

**Comment by NickM-27** [comment]
that is incorrect and will cause issues in 0.13, the entire config directory needs to be mapped. In your case it would need to be

`/dockershare/frigate/config/:/config/`

**Comment by gnlgrim** [comment]
@NickM-27 I am on 0.13. It refused to start with the config you posted.
I had to add it like this

/dockershare/frigate/config/config.yml:/config/config.yml

Otherwise frigate refused to start

**Comment by NickM-27** [comment]
it would be best to figure out why, if I had to guess it is because of the database existing in both places. There is a reason that the docs recommend this, and unless you have the database mapped to a different location your events may be deleted every time you pull the docker image

**Comment by briodan** [comment]
> it would be best to figure out why, if I had to guess it is because of the database existing in both places. There is a reason that the docs recommend this, and unless you have the database mapped to a different location your events may be deleted every time you pull the docker image

The reason is Frigate does not support the database being on a network share at least for me.

The initial problem that they tried to fix in 0.13 was that the database used to live under the media folder, with the introduction on network share support natively in HA this introduced issues specifically for the HA add-on. I think the fix was done primarily to simplify and enhance the ability to run frigate as an HA add-on, which to a point I get trying to make things easier. 

However I feel it can affect a number of other users negatively like myself. I don't have a USB coral, I have a PCIE one, due to the fact that's the only one I could get. Getting a PCIE device passed through HA to the add-on is not straight forward and more likely to break in the future. So I'm running frigate in its own separate LXC container. I also don't care about keeping the event database as I use frigate only for object detection and quick viewing of videos in HA notifications. I could lose the entire DB on a daily basis and would not care. So the only thing I care about is the config file, which is why it mapped to a network share that gets back-ed up regularly.

It might have been more elegant to setup a new folder for the database but I understand also why that might not have been a choice.

**Comment by NickM-27** [comment]
> The initial problem that they tried to fix in 0.13 was that the database used to live under the media folder, with the introduction on network share support natively in HA this introduced issues specifically for the HA add-on. I think the fix was done primarily to simplify and enhance the ability to run frigate as an HA add-on, which to a point I get trying to make things easier.

To be clear I am one of the devs. 

> It might have been more elegant to setup a new folder for the database but I understand also why that might not have been a choice.

There was no other choice due to addon limitations

**Comment by blakeblackshear** [comment]
You also still have full flexibility to put the config file and the db wherever you want. All that changed was the default location. You can override it if needed.
